baseclasses/BaseRegTest.py

```diff
@@ -39,9 +41,14 @@ def __init__(self, ref_file, train=False, comm=None, check_arch=False):
         self.ref_file = ref_file
         if comm is not None:
             self.comm = comm
+            self.rank = self.comm.rank
```

Comment:
Doesn't this already imply we are passing in an MPI object? Would it make sense to rearrange this like:
```
self.ref_file = ref_file
if MPI is None:
    print("No mpi4py dependency, comm input is ignored")
    self.comm = None
    self.rank = 0
else:
    if comm is not None:
        self.comm = comm
    else:
        self.comm = MPI.COMM_WORLD
    self.rank = self.comm.rank
```
Otherwise, we can add an `isInstance` check to comm. We can (and should?) keep the code failing if the user inadvertently provides e.g. `comm="hello world"`, but the error should be handled more explicitly.

Reply:
Lol I also realize this test class should be used only for developers adding new tests, but still I think we can handle it differently?

Reply:
In general we don't really do type checking in Python since it's a dynamically typed language. We can for example look into using [mypy](https://github.com/python/mypy) to do static type checking, but that would be a pretty substantial undertaking. In general, we rely on documentation and users being careful to not pass wrong datatypes, and usually the errors are pretty obvious.

I will change the logic to how you have it there since it's a bit better.